src/main/java/com/github/slem1/await/HttpService.java

```diff
@@ -44,17 +70,26 @@ public String toString() {
 
     @Override
     public void execute() throws ServiceUnavailableException {
-
         try {
-
-            HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
-            urlConnection.setRequestProperty("method", "GET");
+            HttpURLConnection urlConnection;
+            if (url.getProtocol().equals("https")) {
+                urlConnection = (HttpsURLConnection) url.openConnection();
+                if (trustAllCerts) {
+                    SSLContext sc = SSLContext.getInstance("SSL");
+                    sc.init(null, getTrustAllCertsTrustManager(), new java.security.SecureRandom());
```

Comment:
Because multiple httpConnection could need the SSL skip mechanism, It may be useful to use a Singleton pattern to initialize the trustmanager once and for all. What do you think about it ?

Reply:
Please check it, would be that solution fine for you?